tests/integration/test_schema.py

```diff
@@ -1008,7 +1008,12 @@ async def assert_schema_versions(client: Client, trail: str, schema_id: int, exp
     """
     res = await client.get(f"/schemas/ids/{schema_id}/versions{trail}")
     assert res.status_code == 200
-    assert [{"subject": e[0], "version": e[1]} for e in expected] == res.json()
+
+    # Schema Registry doesn't return an ordered list, Karapace does.
+    # Need to check equality ignoring ordering.
+    assert len(res.json()) == len(expected)
+    for e in ({"subject": e[0], "version": e[1]} for e in expected):
```

Comment:
Please change the signature of the function to receive the dictionary values. This will then be:
```python
assert expected == sorted(res.json(), key=lambda result: result["version"])
```

Reply:
The called would then need to create the `dict` always? I think it's easier to write tuples - no need to repeat the keys all the time? What would be the benefit of passing a `dict`?

Also, I think `key` in the `lambda` above should take into account also `subject`. The list in Karapace is ordered like this `key=lambda s: (s["subject"], s["version"]`.

Reply:
> The called would then need to create the dict always?

Yes. I don't understand if the problem you're pointing out is the instantiation itself, because this loop is instantiating the dicts anyways.

> I think it's easier to write tuples - no need to repeat the keys all the time?

Both are extremely easy to read and write:
```python
[
  ("subject1", 1),
  ("subject1", 2),
  ("subject1", 3),
]
```
```python
[
  {"subject": "subject1", "version": 1},
  {"subject": "subject1", "version": 2},
  {"subject": "subject1", "version": 3},
]
```

> What would be the benefit of passing a dict?

It is the actual expected result, not a list of tuples.

> Also, I think key in the lambda above should take into account also subject. The list in Karapace is ordered like this key=lambda s: (s["subject"], s["version"].

Sure, that was just an example to remove the `o^2` complexity and streamline the code

Reply:
I'm leaving like it is, if it causes problems, can be changed later. The effective functionality is the same in both approaches.